# Práctica 1.1 — Probabilidad y Estadística
## Análisis de la evolución poblacional mundial

## Objetivo

Usar un dataset real para probar que el entorno (Pandas, NumPy, Jupyter) funciona y sacar algunas estadísticas básicas.

## Actividad 1 — Identificación de la fuente

Datos de **Our World in Data (OWID)**, fuente original **United Nations, World Population Prospects (2024)**.

Cita:

> UN, World Population Prospects (2024) – processed by Our World in Data. "Annual change in population – UN WPP" [dataset]. United Nations, "World Population Prospects"; United Nations, "World Population Prospects - Interim Update" [original data].

### Preguntas

1. ¿Quién produjo originalmente los datos? La ONU (World Population Prospects 2024).
2. ¿Qué organización los procesa? Our World in Data.
3. ¿Qué representa la variable? El cambio anual de población (nacimientos, defunciones y migración).

## Actividad 2 — Carga y exploración de datos

### 2.1 Importar las bibliotecas

In [1]:
import pandas as pd
import numpy as np

### 2.2 Cargar el archivo CSV

In [2]:
df = pd.read_csv("../data/raw/annual-population-growth/annual-population-growth.csv")

### 2.3 Explorar el DataFrame

In [3]:
# Primeras filas
df.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


In [4]:
# Dimensiones del DataFrame
df.shape

(38400, 5)

In [5]:
# Nombres de columnas
df.columns.tolist()

['Entity',
 'Code',
 'Year',
 'Annual change in population',
 'Annual population change (Projected)']

In [6]:
# Tipos de datos
df.dtypes

Entity                                      str
Code                                        str
Year                                      int64
Annual change in population             float64
Annual population change (Projected)    float64
dtype: object

In [7]:
# Estadísticas descriptivas
df.describe()

,Year,Annual change in population,Annual population change (Projected)
count,38400.000000,1.868800e+04,1.971200e+04
mean,2025.500000,2.063019e+06,8.375481e+05
std,43.300872,9.440859e+06,5.816745e+06
min,1951.000000,-3.315935e+06,-2.291035e+07
25%,1988.000000,1.781000e+03,-9.742750e+03
50%,2025.500000,4.689800e+04,9.350000e+01
75%,2063.000000,3.021128e+05,1.411160e+05
max,2100.000000,9.337137e+07,7.206736e+07


In [8]:
# Cantidad de valores faltantes por columna
df.isnull().sum()

Entity                                      0
Code                                     1200
Year                                        0
Annual change in population             19712
Annual population change (Projected)    18688
dtype: int64

In [9]:
# Cantidad de entidades diferentes
df["Entity"].nunique()

256

### Preguntas

1. ¿Cuántas filas y columnas tiene el dataset? 38,400 filas y 5 columnas.
2. ¿Qué tipo de datos contiene `Year`? int64.
3. ¿Existen valores faltantes? Sí, en `Code` (agregados sin código ISO) y en las dos columnas de cambio de población (una cubre histórico, la otra proyección, así que se completan entre sí).
4. ¿Cuántas entidades diferentes contiene? 256 (países, regiones, agrupaciones y "World").

## Actividad 3 — Estadística descriptiva

Usamos solo los datos históricos (`Annual change in population`). Se toma el año 2020 y se calculan estadísticos sobre todas las entidades ese año.

In [10]:
anio_seleccionado = 2020
df_anio = df[df["Year"] == anio_seleccionado][["Entity", "Year", "Annual change in population"]]
df_anio.head()

,Entity,Year,Annual change in population
69,Afghanistan,2020,1212852.0
219,Africa (UN),2020,32815675.0
369,Albania,2020,-13061.0
519,Algeria,2020,747543.0
669,American Samoa,2020,-451.0


In [11]:
cambio_poblacion = df_anio["Annual change in population"]

media = cambio_poblacion.mean()
mediana = cambio_poblacion.median()
varianza = cambio_poblacion.var()
desviacion_estandar = cambio_poblacion.std()
minimo = cambio_poblacion.min()
maximo = cambio_poblacion.max()

print(f"Media: {media:,.2f}")
print(f"Mediana: {mediana:,.2f}")
print(f"Varianza: {varianza:,.2f}")
print(f"Desviación estándar: {desviacion_estandar:,.2f}")
print(f"Mínimo: {minimo:,.2f}")
print(f"Máximo: {maximo:,.2f}")

Media: 2,124,908.85
Mediana: 34,293.50
Varianza: 90,687,335,031,103.64
Desviación estándar: 9,522,989.82
Mínimo: -494,016.00
Máximo: 75,707,590.00


In [12]:
# chequeo rápido con numpy
print(np.mean(cambio_poblacion), np.std(cambio_poblacion, ddof=1))

2124908.84765625 9522989.815761836


In [13]:
# Entidades con el valor mínimo y máximo en 2020
entidad_minimo = df_anio.loc[df_anio["Annual change in population"].idxmin(), "Entity"]
entidad_maximo = df_anio.loc[df_anio["Annual change in population"].idxmax(), "Entity"]

print("Entidad con el valor mínimo:", entidad_minimo)
print("Entidad con el valor máximo:", entidad_maximo)

Entidad con el valor mínimo: Venezuela
Entidad con el valor máximo: World


### Preguntas

1. ¿Media? ~2,124,909.
2. ¿Mediana? ~34,293 (mucho más baja que la media, por los agregados grandes como "World").
3. ¿Desviación estándar? ~9,522,990, bastante alta por la diferencia de tamaño entre países/agregados.
4. ¿Mínimo y máximo? Mínimo: Venezuela (caída fuerte por la crisis migratoria). Máximo: World (es el agregado global).

## Actividad 4 — Comprobación del entorno

In [14]:
import sys

print("Python:", sys.version)
print("Pandas:", pd.__version__)
print("NumPy:", np.__version__)

Python: 3.12.10 (tags/v3.12.10:0cc8128, Apr  8 2025, 12:21:36) [MSC v.1943 64 bit (AMD64)]
Pandas: 3.0.5
NumPy: 2.5.2


Coincide con lo que está en `requirements.txt`. Se corrió todo de nuevo desde cero (Restart & Run All) y no hubo errores.

## Resultados

- Fuente y cita de los datos (OWID / UN WPP 2024)
- DataFrame cargado y explorado (shape, columnas, tipos, nulos, entidades)
- Estadísticos del cambio de población en 2020 (media, mediana, varianza, std, min, max)
- Versiones de Python/Pandas/NumPy

No se hicieron gráficas ni análisis de proyecciones/temporal — eso queda para prácticas siguientes.